### ISSUE TITLE
Error in function argument renamed destructuring with type

### ISSUE THREAD

**Issue by ghost** [palantir/tslint#3761]
### Bug Report

- __TSLint version__:  5.9.1
- __TypeScript version__: 2.7.2
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
interface SessionMeta {
  // ...
}

this.sessionService
  .getSessions()
  .subscribe( ( { meta: newMeta }:  { meta: SessionMeta } ) => {
    //            |
    //      error points to here

    this.setCount(newMeta.match_count);
  });
```

with `tslint.json` configuration:

```json
{
  "no-use-before-declare": true,
}
```

#### Actual behavior
`ERROR: ...: variable 'meta' used before declaration`

#### Expected behavior
I think, the syntax is valid, the compiled code is good, the expected. You can see a similar example here:
https://github.com/Microsoft/TypeScript-Handbook/blob/master/pages/Variable%20Declarations.md#property-renaming

**Comment by adrienboulle** [comment]
I also have this problem, syntax is fine, I think this is a false positive

**Comment by JoshuaKGoldberg** [comment]
Looks like this was fixed by #3849. 🎉
